folio-service-tools-spring-dev/src/main/java/org/folio/spring/tools/kafka/KafkaAdminService.java

```diff
@@ -42,6 +45,15 @@ public void createTopics(String tenantId) {
     kafkaAdmin.initialize();
   }
 
+  public void deleteTopicsByTenant(String tenantId) {
+    List<String> topicsToDelete = kafkaProperties.getTopics().stream()
+                                                             .filter(t -> t.getName().contains("." + tenantId + "."))
+                                                             .map(FolioKafkaProperties.KafkaTopic::getName)
+                                                             .toList();
```

Comment:
```suggestion
    List<String> topicsToDelete = kafkaProperties.getTopics().stream()
      .map(FolioKafkaProperties.KafkaTopic::getName)
      .filter(name -> name.startsWith(FolioEnvironment.getFolioEnvName() + "." + tenantId + "."))
      .toList();
```
You could use ENV variable + tenantId to identify the start of the topic name.

Reply:
But in the same situation, you will delete all topics that have tenant id in their name. But you should delete only topics that the current module is creating on startup.

Reply:
Use 
`var configTopics = kafkaProperties.getTopics();` to get list of topics
and use 
`getTenantTopicName(topic.getName(), tenantId)` to get full topic name for current tenant